internal/pkg/template/diff/override.go

```diff
@@ -40,58 +43,185 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
+type intrinsicFuncMatcher struct{}
+
+// match returns true if from and to node represent the same intrinsic function.
 // Example1: "!Ref" and "Ref:" will return true.
-// Example2: "!Ref" and "!Ref" will return false because they are written in the same form (i.e. short).
+// Example2: "!Ref" and "!Ref" will return true.
 // Example3: "!Ref" and "Fn::GetAtt:" will return false because they are different intrinsic functions.
-// For more on intrinsic functions and full/short forms, read https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html.
-func (_ *intrinsicFuncFullShortFormConverter) match(from, to *yaml.Node, _ string, _ overrider) bool {
+// Example4: "!Magic" and "Fn::Magic" will return false because they are not intrinsic functions.
+func (_ *intrinsicFuncMatcher) match(from, to *yaml.Node, _ string, _ overrider) bool {
 	if from == nil || to == nil {
 		return false
 	}
-	if (from.Kind == to.Kind) || (from.Kind != yaml.MappingNode && to.Kind != yaml.MappingNode) {
-		// A full/short form conversion always involve at least one mapping node.
+	fromFunc, toFunc := intrinsicFuncName(from), intrinsicFuncName(to)
+	if _, ok := intrinsicFunctions[fromFunc]; !ok {
 		return false
 	}
-	var fullFormNode, shortFormNode *yaml.Node
-	if from.Kind == yaml.MappingNode {
-		fullFormNode, shortFormNode = from, to
-	} else {
-		fullFormNode, shortFormNode = to, from
+	if _, ok := intrinsicFunctions[toFunc]; !ok {
+		return false
 	}
-	if len(fullFormNode.Content) != 2 {
-		// The full form mapping node always contain only one child node, whose key is the func name in full form.
-		// Read https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html.
+	return fromFunc == toFunc
+}
+
+// intrinsicFuncMatcher matches and parses two intrinsic function nodes written in different form (full/short).
+type intrinsicFuncMapTagConverter struct {
+	intrinsicFunc intrinsicFuncMatcher
+}
+
+// match returns true if from and to node represent the same intrinsic function written in different (full/short) form.
+// Example1: "!Ref" and "Ref:" will return true.
+// Example2: "!Ref" and "!Ref" will return false because they are written in the same form (i.e. short).
+// Example3: "!Ref" and "Fn::GetAtt:" will return false because they are different intrinsic functions.
+// For more on intrinsic functions and full/short forms, read https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ToJsonString.html.
+func (converter *intrinsicFuncMapTagConverter) match(from, to *yaml.Node, key string, overrider overrider) bool {
+	if !converter.intrinsicFunc.match(from, to, key, overrider) {
 		return false
 	}
-	return eqIntrinsicFunc(intrinsicFuncFullFormName(fullFormNode), shortFormNode.Tag)
+	// Exactly one of from and to is full form.
+	return (from.Kind == yaml.MappingNode || to.Kind == yaml.MappingNode) && (from.Kind != to.Kind)
```

Comment:
is there a case you know of where one of these would _not_ be a `MappingNode` but also `from.Kind != to.Kind`?

Reply:
Currently, I don't think there is! If a function is to be full-form, it must have been written as `Fn::FuncName:<something>`, which is innately a map, no matter what type of data `<something>` is.